replication-api/src/main/java/com/connexta/replication/queue/Queue.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.queue;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * A queue defines all possible operations that can be performed. Queues are specific to a given
+ * site and will hold tasks which will specify intel that either needs to be harvested from,
+ * retrieved from, or replicated to. For example:
+ *
+ * <ul>
+ *   <li>a Tactical Ion or DIB site we replicate from/to
+ *   <li>a DDF-based site we are harvesting from or replicating to
+ *   <li>a Regional Ion site to which we are replicating our local intel to
+ *   <li>the local Ion site (typically will be used for delete operations only)
```

Comment:
Yes. However, I am trying very hard not to use the term `node` since a site will be composed of multiple nodes since we are cloud based. So the local site is the site we are deployed to be part of.